utils/python/CIME/create_test.py

```diff
@@ -95,17 +98,33 @@ def __init__(self, test_names,
         else:
             self._parallel_jobs = parallel_jobs
 
-        if (self._compare or self._generate):
+        self._baseline_cmp_name = None
+        self._baseline_gen_name = None
 
+        if (self._compare or self._generate):
+            logging.warn("compare is %s generate is %s"%(self._compare,self._generate))
             # Figure out what baseline name to use
             if (baseline_name is None):
+                if(self._compare is not None and self._compare == str(self._compare)):
```

Comment:
_compare and _generate should always be booleans. If the incoming compare argument is a string, set self._compare to True and self._baseline_cmp_name to compare argument.